lib/install.js

```diff
@@ -594,35 +582,27 @@ function isUpToDate(url, requestOpts, hash, cb) {
   if (!hash) {
     return cb(null, false);
   }
-
-  var query = merge({}, requestOpts, {
-    url: url,
+  const options = merge({}, requestOpts,{
+    timeout: 10000,    
+    retry: {
+      limit:3,      
+    },
     headers: {
-      "If-None-Match": '"' + hash + '"',
+      "If-None-Match": `"${hash}"`,
     },
   });
-
-  var req = request.get(query);
-  req
-    .on("response", function (res) {
-      req.abort();
-
-      if (res.statusCode === 304) {
-        return cb(null, true);
-      }
-
-      if (res.statusCode !== 200) {
-        return cb(
-          new Error(
-            "Could not request headers from " + url + ": ",
-            res.statusCode
-          )
-        );
+  (async () => {
+      try {
+          const res = await got(url, options);
+          if (res.statusCode === 304) {
+            return cb(null, true);
+          }
+          if(res.statusCode !== 200) {
```

Comment:
I guess if status code is not 200 for throws an error so it cannot be ever reached